File: client_wpf/Xgame.GamePart/Tmpl/Type/XlsxFloat.cs
using System;

namespace Xgame.GamePart.Tmpl.Type
{
    /// <summary>
    /// Xlsx 浮点型
    /// </summary>
    public class XlsxFloat : PrimitiveTypeCol<float>
    {
        #region 类构造器
        /// <summary>
        /// 类默认构造器
        /// </summary>
        public XlsxFloat()
        {
        }

        /// <summary>
        /// 类参数构造器
        /// </summary>
        /// <param name="val"></param>
        public XlsxFloat(float val)
        {
            this.Val = val;
        }
        #endregion

        // @Override
        protected override void ReadImpl(XlsxRowReadStream stream)
        {
            this.Val = stream.ReadFloat();
        }
    }
}

File: game_part/src/main/java/com/game/part/tmpl/type/XlsxFloat.java
package com.game.part.tmpl.type;

import com.game.part.tmpl.XSSFRowReadStream;

/**
 * Excel Float 字段
 * 
 * @author hjj2019
 * @since 2015/2/23
 * 
 */
public class XlsxFloat extends PrimitiveTypeCol<Float> {
    /**
     * 类默认构造器
     *
     */
    public XlsxFloat() {
        super();
    }

    /**
     * 类参数构造器
     *
     * @param nullable
     *
     */
    public XlsxFloat(boolean nullable) {
        super(nullable);
    }

    /**
     * 类参数构造器
     *
     * @param nullable
     * @param defaultVal
     *
     */
    public XlsxFloat(boolean nullable, float defaultVal) {
        super(nullable, defaultVal);
    }

    /**
     * 类参数构造器
     *
     * @param defaultVal
     *
     */
    public XlsxFloat(float defaultVal) {
        super(defaultVal);
    }

    @Override
    protected void readImpl(XSSFRowReadStream fromStream) {
        if (fromStream != null) {
            super.setObjVal(fromStream.readFloat());
        }
    }
}
